addons/jobs/jobs-management-common/src/main/java/org/kie/kogito/jobs/management/RestJobsService.java

```
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.kogito.jobs.management;
+
+import java.net.URI;
+import java.util.Objects;
+
+import org.kie.kogito.jobs.JobsService;
+import org.kie.kogito.jobs.ProcessInstanceJobDescription;
+import org.kie.kogito.jobs.api.URIBuilder;
+
+public abstract class RestJobsService implements JobsService {
+
+    public static final String JOBS_PATH = "/jobs";
+
+    private URI jobsServiceUri;
+    private String callbackEndpoint;
+
+    public RestJobsService(String jobServiceUrl, String callbackEndpoint) {
+        this.jobsServiceUri = Objects.nonNull(jobServiceUrl) ? buildJobsServiceURI(jobServiceUrl) : null;
```

Wouldn't it have more sense to thrown an exception in case jobServiceUrl is null?
From my point of view it is not valid value and we should fail fast.

---

That is a point but when initializing the runtime with quarkus, in some step it instantiates the bean on CDI using the default constructor, with no args and in this case throwing an exception will break the startup, I tested it =/ 
See https://quarkus.io/guides/cdi-reference on item 5.4

---

ok